### PR TITLE
Add @layer support

### PR DESCRIPTION
**Pull request by aspirisen**
## Description
Fixes: https://github.com/wayfair/rainbow-sprinkles/issues/146

Add `@layer` support

## Type of Change

- [ ] Bug Fix
- [x] New Feature
- [ ] Breaking Change
- [ ] Refactor
- [ ] Documentation
- [ ] Other (please describe)

## Checklist

- [x] I have read the [contributing guidelines](https://github.com/wayfair/rainbow-sprinkles/blob/main/CONTRIBUTING.md)
- [x] Existing issues have been referenced (where applicable)
- [x] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [x] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [x] All new and existing tests pass

### CODE REVIEW

File: packages/rainbow-sprinkles/src/defineProperties.ts [hunk 1]
```diff
@@ -133,13 +133,18 @@ export type OptionsBoth<
   shorthands?: Shorthands;
 };
 
+export interface CommonOptions {
+  '@layer'?: string;
+}
```
Comment (roginfarrer) [inline packages/rainbow-sprinkles/src/defineProperties.ts:138]: Let's use this in `createStyles` and `createStaticStyles`. Move to the `types.ts` file if needed.

**Review by roginfarrer — CHANGES_REQUESTED** [review]
Looking good. Can we add tests for `createStyles` and `createStaticStyles`? To validate the object that we're passing to `style()`, not the actual CSS that's returned.

It would also be great to add a changeset and to update the package readme to add docs for using layers.

**Comment by aspirisen** [comment]
Hey @roginfarrer 
I've made appropriate changes, could you please take a look again

File: .changeset/selfish-yaks-destroy.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'rainbow-sprinkles': minor
+---
+
+Add CSS @layer support
```
Comment (roginfarrer) [inline .changeset/selfish-yaks-destroy.md:5]:
```suggestion
Support assigning properties to layers via `@layer` option on `defineProperties`

**Example usage:**

```ts
// sprinkles.css.ts
import { defineProperties } from 'rainbow-sprinkles';
import { layer } from '@vanilla-extract/css';

export const sprinklesLayer = layer();

const properties = defineProperties({
  '@layer': sprinklesLayer,
  // etc.
});
```
```

File: packages/rainbow-sprinkles/README.md [hunk 3]
```diff
@@ -160,6 +160,44 @@ function App() {
...
+  staticProperties: {
+    display: ['block', 'flex', 'inline-block', 'inline-flex'],
+    border: {
+      '1x': '1px',
+      '2x': '2px',
+      '3x': '3px',
+    },
+  },
+});
+```
```
Comment (roginfarrer) [inline packages/rainbow-sprinkles/README.md:199]:
```suggestion
import { defineProperties } from 'rainbow-sprinkles';

export const responsiveLayer = layer('responsive');

const properties = defineProperties({
  '@layer': sprinklesLayer
  // etc.
});
```
```

**Review by roginfarrer — APPROVED** [review]

**Pull request merged**